src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/PInvokeMarshal.Windows.cs

```diff
@@ -90,5 +91,68 @@ public static unsafe int GetCharCount(byte* multiByteStr, int multiByteLen)
             return Interop.Kernel32.MultiByteToWideChar(Interop.Kernel32.CP_ACP, 0, multiByteStr, multiByteLen, default(char*), 0);
         }
         #endregion
+
+        public static unsafe object InitializeCustomMarshaller(RuntimeTypeHandle pParameterType, RuntimeTypeHandle pMarshallerType, string cookie, delegate*<string, object> getInstanceMethod)
+        {
+            if (getInstanceMethod == null)
+            {
+                throw new ApplicationException();
+            }
+
+            var marshaller = s_customMarshallersTable.GetOrAdd(new CustomMarshallerKey(pParameterType, pMarshallerType, cookie, getInstanceMethod));
+            if (marshaller == null)
+            {
+                throw new ApplicationException();
+            }
+
+            return marshaller;
+        }
+
+        private static CustomMarshallerTable s_customMarshallersTable = new CustomMarshallerTable();
```

Comment:
This static will pull in the `CustomMarshallerTable` into the typical all (static fields in the given type cannot be trimmed independently). The static should be on `CustomMarshallerTable` to keep it separate.